cmd/kubeadm/app/cmd/phases/workflow/runner.go

```diff
@@ -198,6 +198,24 @@ func (e *Runner) Run(args []string) error {
 		return err
 	}
 
+	// precheck phase dependencies before actual execution
+	visited := map[string]struct{}{}
+	err = e.visitAll(func(p *phaseRunner) error {
+		if run, ok := phaseRunFlags[p.generatedName]; !run || !ok {
+			return nil
+		}
+		for _, dep := range p.Phase.Dependencies {
+			if _, ok := visited[dep]; !ok {
+				return errors.Errorf("unresolved dependencies: missing %q phase which is needed by %q phase", dep, p.Phase.Name)
```

Comment:
we could add all the missing dependencies in a new slice and print it with %v, then return a single error

Reply:
Updated, the output will be:
```sh
unresolved dependencies:
	missing [bootstrap-token upload-certs] phase(s) needed by "show-join-command" phase
```